lnbits/core/tasks.py

```diff
@@ -1,20 +1,103 @@
 import asyncio
-from typing import Dict
+from typing import Dict, Optional
 
 import httpx
 from loguru import logger
 
+from lnbits.settings import get_wallet_class, settings
 from lnbits.tasks import SseListenersDict, register_invoice_listener
 
 from . import db
-from .crud import get_balance_notify, get_wallet
+from .crud import get_balance_notify, get_total_balance, get_wallet
 from .models import Payment
-from .services import websocketUpdater
+from .services import switch_to_voidwallet, websocketUpdater
 
 api_invoice_listeners: Dict[str, asyncio.Queue] = SseListenersDict(
     "api_invoice_listeners"
 )
 
+killswitch: Optional[asyncio.Task] = None
+watchdog: Optional[asyncio.Task] = None
+
+
+async def register_killswitch():
+    """
+    Registers a killswitch which will check lnbits-status repository
+    for a signal from LNbits and will switch to VoidWallet if the killswitch is triggered.
+    """
+    logger.debug("Starting killswitch task")
+    global killswitch
+    killswitch = asyncio.create_task(killswitch_task())
+
+
+async def unregister_killswitch():
+    """
+    Unregisters a killswitch taskl
+    """
+    global killswitch
+    if killswitch:
+        logger.debug("Stopping killswitch task")
+        killswitch.cancel()
+
+
+async def killswitch_task():
+    while True:
+        if settings.lnbits_killswitch:
+            with httpx.Client() as client:
```

Comment:
- only execute if active wallet is NOT `VoidWallet`
- otherwise it will keep switching to `VoidWallet`:
```py
2023-05-25 14:23:34.29 | ERROR | lnbits.core.tasks:killswitch_task:53 | Switching to VoidWallet. Killswitch triggered.
2023-05-25 14:23:44.52 | ERROR | lnbits.core.tasks:killswitch_task:53 | Switching to VoidWallet. Killswitch triggered.
2023-05-25 14:23:54.75 | ERROR | lnbits.core.tasks:killswitch_task:53 | Switching to VoidWallet. Killswitch triggered.
2023-05-25 14:24:04.98 | ERROR | lnbits.core.tasks:killswitch_task:53 | Switching to VoidWallet. Killswitch triggered.
2023-05-25 14:24:15.21 | ERROR | lnbits.core.tasks:killswitch_task:53 | Switching to VoidWallet. Killswitch triggered.
2023-05-25 14:24:25.45 | ERROR | lnbits.core.tasks:killswitch_task:53 | Switching to VoidWallet. Killswitch triggered.
```